subsys/net/lib/wifi_provision/certs/server_certificate.pem

```diff
@@ -0,0 +1,12 @@
+-----BEGIN CERTIFICATE-----
```

Comment:
hmm, not sure. I guess its more convenient if the application owns the certificates. I can create a Kconfig that allows the user to specify an app folder that the credentials needs to be included in.

'CONFIG_MQTT_HELPER_CERTIFICATES_FOLDER'

Reply:
Better is to application load the certificates to store, and then just load them when creating the socket. Then you could just pass the tag list to the library instead of the certificate folder etc.

Reply:
That means that all samples that include the wifi provision library needs to also include logic to convert and load certs. Instead of the library doing this.

Reply:
I don't think there's anything wrong in expecting that application deals with its own sensitive data (keys/certs).

But on the other hand, the MQTT helper library approach seems as a good tradeoff here, as it still allows the app to handle certificates while offering some convenience functions to provision certs on the app behalf, IMO we could go that way.

My major concern here was that the credentials were hardcoded within the library, which shouldn't be the case no matter what.

Reply:
Yeah, in a production scenario creds are provisioned as a separate stage to protected storage and referenced using a sec tag in the actual app. I plan to add docs for this, there is also a cmake warning that warns users about the cert use in its current state.

Reply:
So it makes sense to be able to disable automatic cert provisioning actually.

Reply:
Added options:

CONFIG_WIFI_PROVISION_SERVER_CERTIFICATE_REGISTER
CONFIG_WIFI_PROVISION_SERVER_CERTIFICATE_FOLDER